python/paddle/fluid/optimizer.py

```diff
@@ -236,7 +236,8 @@ def create_optimization_pass(self,
 
             # Get custom finish ops for subclasses
             # FIXME: Need to fix this once we figure out how to handle dependencies
-            self._finish_update(loss.block)
+            self._finish_update(loss.block,
+                                [p[0] for p in parameters_and_grads])
```

Comment:
`_finish_update` involves defining operators, so it should be in `with  param_and_grad[0].block.program.optimized_guard(param_and_grad[0]):`.